src/main/java/core/basesyntax/model/AbstractOperation.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.model;
+
+import core.basesyntax.db.Warehouse;
+import core.basesyntax.model.entities.Product;
+
+public abstract class AbstractOperation<T extends Product> implements Executable<T> {
+    protected Warehouse<T> warehouse;
```

Comment:
let's not have this field here (and in the subclasses as well) - we can address the static storage directly, without creating its instance

Reply:
Operations need to know where to execute, the link to this warehouse is provided by the matching service, which is dependent on the exact product sub-type. Static storage is useless, since every service has its own storage with the specific product

Reply:
the storage should be static because we want to imitate the work with the real database

services are having access to the same db/storage, they should not have their own instance of the storage